numba/cuda/simulator/cudadrv/devicearray.py

```diff
@@ -30,23 +30,54 @@ class FakeShape(tuple):
     indexing, similar to the shape in CUDA Python. (Numpy shape arrays allow
     negative indexing)
     '''
+
     def __getitem__(self, k):
         if isinstance(k, int) and k < 0:
             raise IndexError('tuple index out of range')
         return super(FakeShape, self).__getitem__(k)
 
 
+class FakeKernelCUDAArray(object):
```

Comment:
This class could do with a docstring explaining why it does what it does, similar to how `FakeCUDAArray` has one.

Reply:
added